lighthouse-core/gather/computed/critical-request-chains.js

```diff
@@ -17,11 +17,15 @@ class CriticalRequestChains extends ComputedArtifact {
    * For now, we use network priorities as a proxy for "render-blocking"/critical-ness.
    * It's imperfect, but there is not a higher-fidelity signal available yet.
    * @see https://docs.google.com/document/d/1bCDuq9H1ih9iNjgzyAL0gpwNFiEP4TZS-YLRp_RuMlc
-   * @param  {any} request
+   * @param {any} request
+   * @param {!WebInspector.NetworkRequest}
    */
-  static isCritical(request) {
+  static isCritical(request, mainResource) {
     const resourceTypeCategory = request._resourceType && request._resourceType._category;
 
```

Comment:
should we add a `console.assert(mainResource, 'mainResource not provided')`?

Reply:
I could wrap it between disable statements? This is just for debugging purposes right?

Reply:
actually instead:

```js
const assert = require('assert');
...
assert.ok(mainResource,  'mainResource not provided');
```